daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/EvLayering.scala

```diff
@@ -70,16 +70,27 @@ final class LayerTransformerEv(
   layerTransformEv: LayerTransformEv,
   maybeLayerCharsetEv: Maybe[LayerCharsetEv],
   maybeLayerLengthKind: Maybe[LayerLengthKind],
-  maybeLayerLengthInBytesEv: Maybe[LayerLengthInBytesEv],
+  maybeLayerLengthEv: Maybe[LayerLengthEv],
   maybeLayerLengthUnits: Maybe[LayerLengthUnits],
   maybeLayerBoundaryMarkEv: Maybe[LayerBoundaryMarkEv],
-  ci: DPathCompileInfo)
-  extends Evaluatable[LayerTransformer](ci)
+  srd: SequenceRuntimeData)
+  extends Evaluatable[LayerTransformer](srd.dpathCompileInfo)
   with NoCacheEvaluatable[LayerTransformer] {
 
+  /**
+   * A layer transformer is never a constant. It always must be evaluated to get
+   * a new one.
+   *
+   * This prevents it being cached/stored as part of the state of the data processor
+   * where it would be shared across threads.
+   *
+   * Otherwise layer transformers couldn't be stateful. But they often need to be stateful.
+   */
+  override protected def isNeverConstant = true
```

Comment:
Could/should this logic be part of `NoCacheEvaluatable`? Or are these different concepts? Seems like if something shouldn't be cached then it also should never be considered constant? It would be nice if mixing in the right things gave the right behavior without having to set other flags.

Reply:
Good point. I'll arrange for NoCacheExecutable to have that behavior, update its scaladoc accordingly, and remove this flag.